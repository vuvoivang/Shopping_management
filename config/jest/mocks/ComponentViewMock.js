import React from 'react';

const ComponentViewMock = () => <div data-dn="ComponentViewMock">ComponentViewMock</div>;

export default ComponentViewMock;
