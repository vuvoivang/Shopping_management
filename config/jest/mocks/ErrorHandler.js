export const skipError = err => console.log(err);
