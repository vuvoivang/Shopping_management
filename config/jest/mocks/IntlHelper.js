import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { IntlProvider, intlShape } from 'react-intl';
import defaultLanguage from '../../../src/localization/en.json';

const intlProvider = new IntlProvider({ locale: 'en', messages: defaultLanguage }, {});
const { intl } = intlProvider.getChildContext();

const nodeWithIntlProp = node => React.cloneElement(node, { intl });

const mountWithIntl = (node, { context, childContextTypes, ...additionalOptions } = {}) =>
  mount(nodeWithIntlProp(node), {
    context: Object.assign({}, { intl }, context),
    childContextTypes: Object.assign({}, { intl: intlShape }, childContextTypes),
    ...additionalOptions
  });

const defaultLanguageState = {
  locale: 'en',
  formattedMessages: {
    en: defaultLanguage
  }
};

//TODO: figure out why this is breaking
const createComponentWithIntl = (children, props = { locale: 'en', messages: defaultLanguage }) => {
  return renderer.create(<IntlProvider {...props}>{children}</IntlProvider>);
};

export { mountWithIntl, defaultLanguage, defaultLanguageState, createComponentWithIntl };
