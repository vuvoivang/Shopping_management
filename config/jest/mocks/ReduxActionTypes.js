const ReduxActionTypes = {
  ROUTER_HISTORY: '@@router/CALL_HISTORY_METHOD',
  AWAIT: 'KC/AWAIT',
  EMPTY: 'EMPTY_ACTION'
};

export { ReduxActionTypes as default };
