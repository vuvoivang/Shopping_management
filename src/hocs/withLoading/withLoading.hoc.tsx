import Loading from 'components/loading/Loading.component';
import React from 'react';
import { useSelector } from 'react-redux';
import { globalSelector } from 'redux/global';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const withLoading = Component => (props: any) => {
  const countLoading = useSelector(globalSelector.getCountLoading);
  return (
    <>
      {countLoading > 0 && <Loading />}
      <Component {...props} />
    </>
  );
};
