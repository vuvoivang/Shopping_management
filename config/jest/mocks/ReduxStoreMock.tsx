import React from 'react';
import configureMockStore from 'redux-mock-store';
import { merge } from 'lodash';
import { Provider } from 'react-redux';
import thunkMiddleware from 'redux-thunk';
import { shallow, mount } from 'enzyme';
import { routerMiddleware } from 'connected-react-router';
import { AppConstant } from '../../../src/constants/app.constant';
import { getAppState } from '../TestHelper';
import { defaultLanguage } from './IntlHelper';
import history from '../../../src/helpers/history.helper';
import { Router } from 'react-router';
import { getLanguageMessages } from '../../../src/utilities/i18n/i18n.utility';
import { IntlProvider } from 'react-intl';

const languageInitialState = {
  [AppConstant.redux.LANGUAGE_STATE]: {
    locale: 'en',
    formattedMessages: {
      en: defaultLanguage
    }
  }
};

export const defaultStore = {
  router: {
    location: {}
  }
};
const { locale, languageMessages } = getLanguageMessages(navigator.language.split(/[-_]/)[0]);

export const shallowWithProvider = children => (store = getMockStore({})) => shallow(<Provider
  store={store}>{children}</Provider>);
export const mountWithProvider = children => (store = getMockStore({})) => mount(<Provider
  store={store}>{children}</Provider>);

export const mountWithProviderAndRouter = children => (store = getMockStore({})) => {
  return mount(
    (
      <Provider store={store}>
        <IntlProvider locale={locale} messages={languageMessages}>
          <Router history={history}>
            {children}
          </Router>
        </IntlProvider>
      </Provider>
    )
  );
}

export const mountWithProviderAndStore = (
  children,
  userType?: string,
  store?: { [key: string]: any },
  middleware = []
) => {
  let mockStore;
  if (!store) {
    mockStore = getMockStore(defaultStore, middleware);
  } else {
    mockStore = getMockStore(store, middleware);
  }
  return mountWithProvider(children)(mockStore);
};

export const getMockStore = (state, middleware = []) => {
  const middlewares = [thunkMiddleware, routerMiddleware(history)].concat(middleware);
  const mockStore = configureMockStore(middlewares);
  const languageAddedState = merge({}, languageInitialState, getAppState(), state);
  return mockStore(languageAddedState);
};

export const getMockStoreWithConfig = ({ state = {}, middleware = [], epic, epics = [] }) => {
  let reduxMiddlewares = [];
  if (middleware && middleware.length) {
    reduxMiddlewares.push(...middleware);
  }

  return getMockStore(state, reduxMiddlewares);
};
