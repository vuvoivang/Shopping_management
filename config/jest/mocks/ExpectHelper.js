import ReduxActionTypes from './ReduxActionTypes';

export const expectActionCalled = (store, actionType, { numberOfCalled = 1, payload, debug = false } = {}) => {
  const actions = store.getActions();
  if (debug) {
    console.log('store mock actions');
    console.log(actions);
  }
  const expectedActions = actions.filter(action => action.type === actionType);
  expect(expectedActions.length).toEqual(numberOfCalled);
  if (payload) {
    expect(payload).toEqual(expectedActions[0].payload);
  }
};

export const expectRouteActionCalledWithURL = (store, URL, { debug = false, actionIndex = -1 } = {}) => {
  const actions = store.getActions();
  if (debug) {
    console.log('store mock actions');
    console.log(actions);
  }

  const routeAction = actionIndex === -1 ? actions.filter(action => action.type === ReduxActionTypes.ROUTER_HISTORY)[0] : store.getActions()[actionIndex];

  expect(routeAction.payload.args[0]).toEqual(URL);
};

export const expectRouteActionCalledWithOptions = (store, options = {}, { debug = false, actionIndex = -1 } = {}) => {
  const actions = store.getActions();
  if (debug) {
    console.log('store mock actions');
    console.log(actions);
  }

  const routeAction = actionIndex === -1 ? actions.filter(action => action.type === ReduxActionTypes.ROUTER_HISTORY)[0] : store.getActions()[actionIndex];

  expect(routeAction.payload.args[0]).toEqual(options);
};

export const expectObjectHasProperties = (object = {}, properties = []) => {
  properties.forEach(property => expect(object.hasOwnProperty(property)).toEqual(true));
};
