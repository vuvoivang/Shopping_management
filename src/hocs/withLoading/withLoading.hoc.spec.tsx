import React from 'react';
import { getMockStore, mountWithProviderAndRouter } from '../../../config/jest/mocks/ReduxStoreMock';
import { withLoading } from './withLoading.hoc';
import About from 'pages/about/About.page';

describe('withLoading', () => {
  const storeWithLoading = {
    globalState: { loadingActions: 1 }
  };
  const storeWithNotLoading = {
    globalState: { loadingActions: 0 }
  };
  const mockStoreLoading = getMockStore(storeWithLoading);
  const mockStoreNotLoading = getMockStore(storeWithNotLoading);

  const setUp = store => {
    const HocWithLoadingComponent = withLoading(About);
    const wrapper = mountWithProviderAndRouter(<HocWithLoadingComponent />)(store);
    return { wrapper };
  };

  it('loading as expect', () => {
    const { wrapper } = setUp(mockStoreLoading);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Loading').exists()).toBeTruthy();
  });
  it('not loading as expect', () => {
    const { wrapper } = setUp(mockStoreNotLoading);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Loading').exists()).toBeFalsy();
  });
});
