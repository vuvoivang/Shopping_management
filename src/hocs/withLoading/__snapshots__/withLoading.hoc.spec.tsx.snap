// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLoading loading as expect 1`] = `ReactWrapper {}`;

exports[`withLoading not loading as expect 1`] = `ReactWrapper {}`;
